tests/FullApplicationTest.php

```diff
@@ -522,6 +522,19 @@ public function testRequestUser()
         $this->assertSame('1234', $response->getContent());
     }
 
+    public function testRequestRouteParam()
+    {
+        $app = new Application();
+
+        $app->get('/{id}', function (Laravel\Lumen\Http\Request $request) {
+            return $request->route('id');
+        });
+
+        $response = $app->handle(Request::create('/4321', 'GET'));
+
+        $this->assertSame('4321', $response->getContent());
```

Comment:
This is what confuses me:

This test fails with the `Call to a member function setParameter() on array` error before my change.
After my change, if the type hint on the Closure is left as `Illuminate\Http\Request` the error no longer happens, but the test fails because `$response->getContent()` returns ''
With the type hint as `Laravel\Lumen\Http\Request` the test passes.

I've been out of PHP for a few years, so I'm new to the Closure handling in PHP. Not sure what I'm missing with why this won't work with the type hint for the parent class.